include/irr/asset/IDescriptorSet.h

```diff
@@ -31,28 +32,29 @@ enum E_IMAGE_LAYOUT : uint32_t
 template<typename BufferType, typename TextureType, typename BufferViewType>
 class IDescriptorSet
 {
-public:
-    struct SDescriptorBufferInfo
-    {
-        core::smart_refctd_ptr<BufferType> buffer;
-        size_t offset;
-        size_t size;//in Vulkan it's called `range` but IMO it's misleading so i changed to `size`
-    };
-    struct SDescriptorImageInfo
+protected:
+    struct SDescriptorInfo
     {
-        SSamplerParams sampler;
-        core::smart_refctd_ptr<TextureType> imageView;
-        //! Irrelevant in OpenGL backend
-        E_IMAGE_LAYOUT imageLayout;
+        core::smart_refctd_ptr<IDescriptor> desc;
+        struct SDescriptorBufferInfo
+        {
+            size_t offset;
+            size_t size;//in Vulkan it's called `range` but IMO it's misleading so i changed to `size`
+        } buffer;
+        struct SDescriptorImageInfo
+        {
+            SSamplerParams sampler;
```

Comment:
you can introduce sampler object (in OpenGL its shareable)

Reply:
resolved